### ISSUE TITLE
[CORAL]SN use stateless in CORAL cluster

### ISSUE THREAD

**Issue by zet809** [xcat2/xcat-core#3681]
To support stateless SN in CORAL cluster.
- [x] Figure out the osimage for general xCAT SN and build diskless image for service node
- [x] Including MOFED for CORAL (Just follow the steps for CN)
- [x] Provision SN as diskless and health check all kinds for services on it
- [x] localdisk investigation (#3860)

After about investigation to work out our gap and requirement.

**Comment by zet809** [comment]
Let's move to next sprint to get detailed requirement.

**Comment by robin2008** [comment]
Create customized osimage for SN stateless as below:
```
# <xCAT data object stanza file>

rhels7.4-Alpha1-ppc64le-netboot-service-mlnx:
    objtype=osimage
    exlist=/opt/xcat/share/xcat/netboot/rh/compute.rhels7.ppc64le.exlist
    imagetype=linux
    osarch=ppc64le
    osdistroname=rhels7.4-Alpha1-ppc64le
    osname=Linux
    osvers=rhels7.4-Alpha1
    otherpkgdir=/install/custom/post/otherpkgs/rhels7.4-Alpha1-service-MLNX/ppc64le
    otherpkglist=/install/custom/install/rh/rhels7.4-Alpha1-service-MLNX/service.rhels7.ppc64le.otherpkgs.pkglist
    permission=755
    pkgdir=/install/rhels7.4-Alpha1/ppc64le
    pkglist=/install/custom/install/rh/rhels7.4-Alpha1-service-MLNX/service.rhels7.ppc64le.pkglist
    postinstall=/install/custom/install/rh/rhels7.4-Alpha1-service-MLNX/service.rhels7.ppc64le.postinstall
    postscripts=servicenode
    profile=service
    provmethod=netboot
    rootimgdir=/install/custom/rhels7.4-Alpha1-ppc64le-netboot-service-mlnx/compute
```

here, `compute.rhels7.ppc64le.exlist` and `service.rhels7.ppc64le.otherpkgs.pkglist` is from `/opt/xcat/share/xcat/install/rh/` directly.

And `service.rhels7.ppc64le.pkglist` is not enough for stateless,  I also merge the package list from 'compute.rhels7.ppc64le.pkglist',  And there might be some package which could be removed from the combined list, but I don't do it yet. I think the key to add the `kernel` package there.

For postinstall,  xCAT don't ship a default postinstall for rhels7 service node,  I just use `/opt/xcat/share/xcat/netboot/rh/compute.rhels7.ppc64le.postinstall`

**Comment by robin2008** [comment]
To package MOFED into image, we need below:
1, put MOFED iso into a specific directory on MN, for example, `/install/mlnxofed/MLNX_OFED_LINUX-4.1-4.0.7.1-rhel7.4alternate-ppc64le.iso`

2, Add below command into postinstall script:
`/install/postscripts/mlnxofed_ib_install -p /install/mlnxofed/MLNX_OFED_LINUX-4.1-4.0.7.1-rhel7.4alternate-ppc64le.iso -m --add-kernel-support --force -end- -i $1 -n genimage`

3, Include IB support packages list file into our package list file
`#INCLUDE:/opt/xcat/share/xcat/ib/netboot/rh/ib.rhels7.ppc64le.pkglist#`

4, `genimage/packimage` again.

**Comment by robin2008** [comment]
Test the provision Today, and found some issues:
1, Failed to get credentials when running remoteshell/servicnode
**Root cause**:  firewalld is not disabled on stateless SN.
workaround: chroot and disable firewalld in rootimg directory.

2, xcatd is not started successfully after stateless SN is provisioned.
**Root cause**:  Not finished yet. I guess that the xcatd is started when cfgloc is not downloaded.
workaround: restart xcatd manually

**Comment by daniceexi** [comment]
Good Job.

But we did not hear that firewall is a not problem for diskless SN.

**Comment by robin2008** [comment]
Adjust the package list and refine the `servicenode` script, we can get a workable stateless SN without any workaround.
The fix is in PR #3859

**Comment by robin2008** [comment]
The final osimage for stateless SN example is in #3860
